### PR TITLE
Refactor federated services tests a bit to move a test that requires no cluster creation to a separate block.

### PR DESCRIPTION
**Pull request by madhusudancs**
Follow up to PR #40769.

cc @kubernetes/sig-federation-pr-reviews

### CODE REVIEW

**Comment by madhusudancs** [comment]
@k8s-bot cvm gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot unit test this

File: test/e2e_federation/federated-service.go [hunk 1]
```diff
@@ -144,6 +155,7 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 				By(fmt.Sprintf("Verified that services were not deleted from underlying clusters"))
 			})
 		})
+
```
Comment (shashidharatd) [inline test/e2e_federation/federated-service.go]: @madhusudancs, i think this is missed out for few delete test cases above.
The complete context for the test case would read like below
"Federated Services [Feature:Federation] With Clusters **Service creation** should be deleted from underlying clusters when OrphanDependents is false"

Comment (madhusudancs) [inline test/e2e_federation/federated-service.go]: But isn't skipping **Service creation** the right thing? "Service creation should be deleted" doesn't make any sense right?

Comment (shashidharatd) [inline test/e2e_federation/federated-service.go]: Sorry for my wording. if you look at Describe clause carefully, you would find delete test cases inside 'Service creation' clause.

Comment (madhusudancs) [inline test/e2e_federation/federated-service.go]: I see your point now. Honestly, I don't know why we have another `Describe` block in there for "Service creation". We don't do it in other tests in Kubernetes also. As I said the organization in this file is absurd. I would just leave refactoring this overall file to another day.

Comment (shashidharatd) [inline test/e2e_federation/federated-service.go]: ok @madhusudancs. SGTM.

**Comment by shashidharatd** [comment]
@k8s-bot unit test this
@k8s-bot cvm gce e2e test this

**Comment by shashidharatd** [comment]
/lgtm

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/40786#-:-KbzMgN5xteEAzJxueus:b-8ks5qa)*

**Comment by madhusudancs** [comment]
Ok, that's weird. Nothing was changed in the last 6 hours. Don't know why the bot thinks so.

**Comment by shashidharatd** [comment]
@madhusudancs, i think you should reapply the approve label

**Comment by shashidharatd** [comment]
@k8s-bot gci gke e2e test this
@k8s-bot gce etcd3 e2e test this
@k8s-bot cri e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot cri e2e test this
@k8s-bot gce etcd3 e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot cri e2e test this

**Pull request merged**